modules/social_features/social_event/src/Access/EventEnrollmentsAccessCheck.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\social_event\Access;
+
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\social_event\Entity\Node\Event;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Implements access checker for enrollments in the event.
+ */
+class EventEnrollmentsAccessCheck implements AccessInterface {
+
+  /**
+   * Check access for enrollments pages in the event.
+   */
+  public function access(Route $route, AccountInterface $account, RouteMatchInterface $route_match): AccessResultInterface {
+    // Skip for administrators.
+    if (
+      $account->hasPermission('bypass node access') ||
+      $account->hasPermission('administer event enrollment entities')
+    ) {
+      return AccessResult::allowed();
+    }
+
+    $node = $route_match->getParameter('node');
+    if (!$node instanceof NodeInterface) {
+      // On views url parameters node can be provided as raw parameter.
```

Comment:
Would it possible to add actually relative paths so that developer can quickly recognize just by reading them ?

Reply:
I think it's bad idea to list here views paths (which are dynamic and I'm pretty sure it hard to do it here).